budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/web/pages/contract/overview/table/ContractOverviewTable.java

```diff
@@ -77,15 +83,49 @@ protected void populateItem(ListItem<DynamicAttributeField> item) {
                         EditContractPage.createParameters(contractId)));
             }
         });
+
         table.add(new ListView<String>("footerRow", model(from(data).getFooter()) ) {
             @Override
             protected void populateItem(ListItem<String> item) {
                 item.add(new Label("footerItem", item.getModelObject()));
             }
         });
+
+        addTableSummaryLabels(table,model(from(data.getContracts())));
         add(table);
     }
 
+    private void addTableSummaryLabels(WebMarkupContainer table, IModel<List<ContractBaseData>> model){
+
+        IModel<ContractTotalData> totalModel = new TotalContractDetailsModel(model);
+
+        // Add empty table cells in the contract attribute columns
+        List<String> contractAttributes = ((TotalContractDetailsModel) totalModel).getContractAttributes();
+
+        table.add(new ListView<String>("emptyRow", contractAttributes) {
+            @Override
+            protected void populateItem(ListItem<String> item) {
+                item.add(new Label("emptyRowText", Model.of("")));
+            }
+        });
+
+        table.add(new MoneyLabel("totalAmount",
+                new TaxBudgetUnitMoneyModel(
+                        new BudgetUnitMoneyModel(model(from(totalModel).getBudget())),
```

Comment:
If you adjust `TotalContractDetailsModel` as I suggested,  you will need to adjust these method calls here too:
`new BudgetUnitMoneyModel(model(from(totalModel.getObject().getBudget()))),`